// @stargazefi/api/lib/generated/cosmos/feegrant/v1beta1/query.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
exports.QueryClientImpl = exports.QueryFeeAllowancesResponse = exports.QueryFeeAllowancesRequest = exports.QueryFeeAllowanceResponse = exports.QueryFeeAllowanceRequest = exports.protobufPackage = void 0;
/* eslint-disable */
var feegrant_1 = require("../../../cosmos/feegrant/v1beta1/feegrant");
var pagination_1 = require("../../../cosmos/base/query/v1beta1/pagination");
var minimal_1 = require("protobufjs/minimal");
exports.protobufPackage = "cosmos.feegrant.v1beta1";
var baseQueryFeeAllowanceRequest = { granter: "", grantee: "" };
exports.QueryFeeAllowanceRequest = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.granter !== "") {
            writer.uint32(10).string(message.granter);
        }
        if (message.grantee !== "") {
            writer.uint32(18).string(message.grantee);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseQueryFeeAllowanceRequest);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.granter = reader.string();
                    break;
                case 2:
                    message.grantee = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseQueryFeeAllowanceRequest);
        if (object.granter !== undefined && object.granter !== null) {
            message.granter = String(object.granter);
        }
        else {
            message.granter = "";
        }
        if (object.grantee !== undefined && object.grantee !== null) {
            message.grantee = String(object.grantee);
        }
        else {
            message.grantee = "";
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseQueryFeeAllowanceRequest);
        if (object.granter !== undefined && object.granter !== null) {
            message.granter = object.granter;
        }
        else {
            message.granter = "";
        }
        if (object.grantee !== undefined && object.grantee !== null) {
            message.grantee = object.grantee;
        }
        else {
            message.grantee = "";
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.granter !== undefined && (obj.granter = message.granter);
        message.grantee !== undefined && (obj.grantee = message.grantee);
        return obj;
    }
};
var baseQueryFeeAllowanceResponse = {};
exports.QueryFeeAllowanceResponse = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.feeAllowance !== undefined) {
            feegrant_1.FeeAllowanceGrant.encode(message.feeAllowance, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseQueryFeeAllowanceResponse);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.feeAllowance = feegrant_1.FeeAllowanceGrant.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseQueryFeeAllowanceResponse);
        if (object.feeAllowance !== undefined && object.feeAllowance !== null) {
            message.feeAllowance = feegrant_1.FeeAllowanceGrant.fromJSON(object.feeAllowance);
        }
        else {
            message.feeAllowance = undefined;
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseQueryFeeAllowanceResponse);
        if (object.feeAllowance !== undefined && object.feeAllowance !== null) {
            message.feeAllowance = feegrant_1.FeeAllowanceGrant.fromPartial(object.feeAllowance);
        }
        else {
            message.feeAllowance = undefined;
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.feeAllowance !== undefined &&
            (obj.feeAllowance = message.feeAllowance
                ? feegrant_1.FeeAllowanceGrant.toJSON(message.feeAllowance)
                : undefined);
        return obj;
    }
};
var baseQueryFeeAllowancesRequest = { grantee: "" };
exports.QueryFeeAllowancesRequest = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.grantee !== "") {
            writer.uint32(10).string(message.grantee);
        }
        if (message.pagination !== undefined) {
            pagination_1.PageRequest.encode(message.pagination, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseQueryFeeAllowancesRequest);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.grantee = reader.string();
                    break;
                case 2:
                    message.pagination = pagination_1.PageRequest.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseQueryFeeAllowancesRequest);
        if (object.grantee !== undefined && object.grantee !== null) {
            message.grantee = String(object.grantee);
        }
        else {
            message.grantee = "";
        }
        if (object.pagination !== undefined && object.pagination !== null) {
            message.pagination = pagination_1.PageRequest.fromJSON(object.pagination);
        }
        else {
            message.pagination = undefined;
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseQueryFeeAllowancesRequest);
        if (object.grantee !== undefined && object.grantee !== null) {
            message.grantee = object.grantee;
        }
        else {
            message.grantee = "";
        }
        if (object.pagination !== undefined && object.pagination !== null) {
            message.pagination = pagination_1.PageRequest.fromPartial(object.pagination);
        }
        else {
            message.pagination = undefined;
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.grantee !== undefined && (obj.grantee = message.grantee);
        message.pagination !== undefined &&
            (obj.pagination = message.pagination
                ? pagination_1.PageRequest.toJSON(message.pagination)
                : undefined);
        return obj;
    }
};
var baseQueryFeeAllowancesResponse = {};
exports.QueryFeeAllowancesResponse = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        for (var _i = 0, _a = message.feeAllowances; _i < _a.length; _i++) {
            var v = _a[_i];
            feegrant_1.FeeAllowanceGrant.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.pagination !== undefined) {
            pagination_1.PageResponse.encode(message.pagination, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseQueryFeeAllowancesResponse);
        message.feeAllowances = [];
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.feeAllowances.push(feegrant_1.FeeAllowanceGrant.decode(reader, reader.uint32()));
                    break;
                case 2:
                    message.pagination = pagination_1.PageResponse.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseQueryFeeAllowancesResponse);
        message.feeAllowances = [];
        if (object.feeAllowances !== undefined && object.feeAllowances !== null) {
            for (var _i = 0, _a = object.feeAllowances; _i < _a.length; _i++) {
                var e = _a[_i];
                message.feeAllowances.push(feegrant_1.FeeAllowanceGrant.fromJSON(e));
            }
        }
        if (object.pagination !== undefined && object.pagination !== null) {
            message.pagination = pagination_1.PageResponse.fromJSON(object.pagination);
        }
        else {
            message.pagination = undefined;
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseQueryFeeAllowancesResponse);
        message.feeAllowances = [];
        if (object.feeAllowances !== undefined && object.feeAllowances !== null) {
            for (var _i = 0, _a = object.feeAllowances; _i < _a.length; _i++) {
                var e = _a[_i];
                message.feeAllowances.push(feegrant_1.FeeAllowanceGrant.fromPartial(e));
            }
        }
        if (object.pagination !== undefined && object.pagination !== null) {
            message.pagination = pagination_1.PageResponse.fromPartial(object.pagination);
        }
        else {
            message.pagination = undefined;
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        if (message.feeAllowances) {
            obj.feeAllowances = message.feeAllowances.map(function (e) {
                return e ? feegrant_1.FeeAllowanceGrant.toJSON(e) : undefined;
            });
        }
        else {
            obj.feeAllowances = [];
        }
        message.pagination !== undefined &&
            (obj.pagination = message.pagination
                ? pagination_1.PageResponse.toJSON(message.pagination)
                : undefined);
        return obj;
    }
};
var QueryClientImpl = /** @class */ (function () {
    function QueryClientImpl(rpc) {
        this.rpc = rpc;
    }
    QueryClientImpl.prototype.FeeAllowance = function (request) {
        var data = exports.QueryFeeAllowanceRequest.encode(request).finish();
        var promise = this.rpc.request("cosmos.feegrant.v1beta1.Query", "FeeAllowance", data);
        return promise.then(function (data) {
            return exports.QueryFeeAllowanceResponse.decode(new minimal_1.Reader(data));
        });
    };
    QueryClientImpl.prototype.FeeAllowances = function (request) {
        var data = exports.QueryFeeAllowancesRequest.encode(request).finish();
        var promise = this.rpc.request("cosmos.feegrant.v1beta1.Query", "FeeAllowances", data);
        return promise.then(function (data) {
            return exports.QueryFeeAllowancesResponse.decode(new minimal_1.Reader(data));
        });
    };
    return QueryClientImpl;
}());
exports.QueryClientImpl = QueryClientImpl;
var globalThis = (function () {
    if (typeof globalThis !== "undefined")
        return globalThis;
    if (typeof self !== "undefined")
        return self;
    if (typeof window !== "undefined")
        return window;
    if (typeof global !== "undefined")
        return global;
    throw "Unable to locate global object";
})();
