// @stargazefi/api/lib/generated/cosmos/authz/v1beta1/query.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
exports.QueryClientImpl = exports.QueryAuthorizationsResponse = exports.QueryAuthorizationsRequest = exports.QueryAuthorizationResponse = exports.QueryAuthorizationRequest = exports.protobufPackage = void 0;
/* eslint-disable */
var authz_1 = require("../../../cosmos/authz/v1beta1/authz");
var pagination_1 = require("../../../cosmos/base/query/v1beta1/pagination");
var minimal_1 = require("protobufjs/minimal");
exports.protobufPackage = "cosmos.authz.v1beta1";
var baseQueryAuthorizationRequest = {
    granter: "",
    grantee: "",
    methodName: ""
};
exports.QueryAuthorizationRequest = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.granter !== "") {
            writer.uint32(10).string(message.granter);
        }
        if (message.grantee !== "") {
            writer.uint32(18).string(message.grantee);
        }
        if (message.methodName !== "") {
            writer.uint32(26).string(message.methodName);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseQueryAuthorizationRequest);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.granter = reader.string();
                    break;
                case 2:
                    message.grantee = reader.string();
                    break;
                case 3:
                    message.methodName = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseQueryAuthorizationRequest);
        if (object.granter !== undefined && object.granter !== null) {
            message.granter = String(object.granter);
        }
        else {
            message.granter = "";
        }
        if (object.grantee !== undefined && object.grantee !== null) {
            message.grantee = String(object.grantee);
        }
        else {
            message.grantee = "";
        }
        if (object.methodName !== undefined && object.methodName !== null) {
            message.methodName = String(object.methodName);
        }
        else {
            message.methodName = "";
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseQueryAuthorizationRequest);
        if (object.granter !== undefined && object.granter !== null) {
            message.granter = object.granter;
        }
        else {
            message.granter = "";
        }
        if (object.grantee !== undefined && object.grantee !== null) {
            message.grantee = object.grantee;
        }
        else {
            message.grantee = "";
        }
        if (object.methodName !== undefined && object.methodName !== null) {
            message.methodName = object.methodName;
        }
        else {
            message.methodName = "";
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.granter !== undefined && (obj.granter = message.granter);
        message.grantee !== undefined && (obj.grantee = message.grantee);
        message.methodName !== undefined && (obj.methodName = message.methodName);
        return obj;
    }
};
var baseQueryAuthorizationResponse = {};
exports.QueryAuthorizationResponse = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.authorization !== undefined) {
            authz_1.AuthorizationGrant.encode(message.authorization, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseQueryAuthorizationResponse);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authorization = authz_1.AuthorizationGrant.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseQueryAuthorizationResponse);
        if (object.authorization !== undefined && object.authorization !== null) {
            message.authorization = authz_1.AuthorizationGrant.fromJSON(object.authorization);
        }
        else {
            message.authorization = undefined;
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseQueryAuthorizationResponse);
        if (object.authorization !== undefined && object.authorization !== null) {
            message.authorization = authz_1.AuthorizationGrant.fromPartial(object.authorization);
        }
        else {
            message.authorization = undefined;
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.authorization !== undefined &&
            (obj.authorization = message.authorization
                ? authz_1.AuthorizationGrant.toJSON(message.authorization)
                : undefined);
        return obj;
    }
};
var baseQueryAuthorizationsRequest = { granter: "", grantee: "" };
exports.QueryAuthorizationsRequest = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.granter !== "") {
            writer.uint32(10).string(message.granter);
        }
        if (message.grantee !== "") {
            writer.uint32(18).string(message.grantee);
        }
        if (message.pagination !== undefined) {
            pagination_1.PageRequest.encode(message.pagination, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseQueryAuthorizationsRequest);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.granter = reader.string();
                    break;
                case 2:
                    message.grantee = reader.string();
                    break;
                case 3:
                    message.pagination = pagination_1.PageRequest.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseQueryAuthorizationsRequest);
        if (object.granter !== undefined && object.granter !== null) {
            message.granter = String(object.granter);
        }
        else {
            message.granter = "";
        }
        if (object.grantee !== undefined && object.grantee !== null) {
            message.grantee = String(object.grantee);
        }
        else {
            message.grantee = "";
        }
        if (object.pagination !== undefined && object.pagination !== null) {
            message.pagination = pagination_1.PageRequest.fromJSON(object.pagination);
        }
        else {
            message.pagination = undefined;
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseQueryAuthorizationsRequest);
        if (object.granter !== undefined && object.granter !== null) {
            message.granter = object.granter;
        }
        else {
            message.granter = "";
        }
        if (object.grantee !== undefined && object.grantee !== null) {
            message.grantee = object.grantee;
        }
        else {
            message.grantee = "";
        }
        if (object.pagination !== undefined && object.pagination !== null) {
            message.pagination = pagination_1.PageRequest.fromPartial(object.pagination);
        }
        else {
            message.pagination = undefined;
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.granter !== undefined && (obj.granter = message.granter);
        message.grantee !== undefined && (obj.grantee = message.grantee);
        message.pagination !== undefined &&
            (obj.pagination = message.pagination
                ? pagination_1.PageRequest.toJSON(message.pagination)
                : undefined);
        return obj;
    }
};
var baseQueryAuthorizationsResponse = {};
exports.QueryAuthorizationsResponse = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        for (var _i = 0, _a = message.authorizations; _i < _a.length; _i++) {
            var v = _a[_i];
            authz_1.AuthorizationGrant.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.pagination !== undefined) {
            pagination_1.PageResponse.encode(message.pagination, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseQueryAuthorizationsResponse);
        message.authorizations = [];
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authorizations.push(authz_1.AuthorizationGrant.decode(reader, reader.uint32()));
                    break;
                case 2:
                    message.pagination = pagination_1.PageResponse.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseQueryAuthorizationsResponse);
        message.authorizations = [];
        if (object.authorizations !== undefined && object.authorizations !== null) {
            for (var _i = 0, _a = object.authorizations; _i < _a.length; _i++) {
                var e = _a[_i];
                message.authorizations.push(authz_1.AuthorizationGrant.fromJSON(e));
            }
        }
        if (object.pagination !== undefined && object.pagination !== null) {
            message.pagination = pagination_1.PageResponse.fromJSON(object.pagination);
        }
        else {
            message.pagination = undefined;
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseQueryAuthorizationsResponse);
        message.authorizations = [];
        if (object.authorizations !== undefined && object.authorizations !== null) {
            for (var _i = 0, _a = object.authorizations; _i < _a.length; _i++) {
                var e = _a[_i];
                message.authorizations.push(authz_1.AuthorizationGrant.fromPartial(e));
            }
        }
        if (object.pagination !== undefined && object.pagination !== null) {
            message.pagination = pagination_1.PageResponse.fromPartial(object.pagination);
        }
        else {
            message.pagination = undefined;
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        if (message.authorizations) {
            obj.authorizations = message.authorizations.map(function (e) {
                return e ? authz_1.AuthorizationGrant.toJSON(e) : undefined;
            });
        }
        else {
            obj.authorizations = [];
        }
        message.pagination !== undefined &&
            (obj.pagination = message.pagination
                ? pagination_1.PageResponse.toJSON(message.pagination)
                : undefined);
        return obj;
    }
};
var QueryClientImpl = /** @class */ (function () {
    function QueryClientImpl(rpc) {
        this.rpc = rpc;
    }
    QueryClientImpl.prototype.Authorization = function (request) {
        var data = exports.QueryAuthorizationRequest.encode(request).finish();
        var promise = this.rpc.request("cosmos.authz.v1beta1.Query", "Authorization", data);
        return promise.then(function (data) {
            return exports.QueryAuthorizationResponse.decode(new minimal_1.Reader(data));
        });
    };
    QueryClientImpl.prototype.Authorizations = function (request) {
        var data = exports.QueryAuthorizationsRequest.encode(request).finish();
        var promise = this.rpc.request("cosmos.authz.v1beta1.Query", "Authorizations", data);
        return promise.then(function (data) {
            return exports.QueryAuthorizationsResponse.decode(new minimal_1.Reader(data));
        });
    };
    return QueryClientImpl;
}());
exports.QueryClientImpl = QueryClientImpl;
var globalThis = (function () {
    if (typeof globalThis !== "undefined")
        return globalThis;
    if (typeof self !== "undefined")
        return self;
    if (typeof window !== "undefined")
        return window;
    if (typeof global !== "undefined")
        return global;
    throw "Unable to locate global object";
})();
