// @stargazefi/api/lib/generated/stargaze/claim/v1beta1/genesis.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
var coin_1 = require("../../../cosmos/base/v1beta1/coin");
var params_1 = require("../../../stargaze/claim/v1beta1/params");
var claim_record_1 = require("../../../stargaze/claim/v1beta1/claim_record");
var minimal_1 = require("protobufjs/minimal");
exports.protobufPackage = "publicawesome.stargaze.claim.v1beta1";
var baseGenesisState = {};
exports.GenesisState = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.moduleAccountBalance !== undefined) {
            coin_1.Coin.encode(message.moduleAccountBalance, writer.uint32(10).fork()).ldelim();
        }
        if (message.params !== undefined) {
            params_1.Params.encode(message.params, writer.uint32(18).fork()).ldelim();
        }
        for (var _i = 0, _a = message.claimRecords; _i < _a.length; _i++) {
            var v = _a[_i];
            claim_record_1.ClaimRecord.encode(v, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseGenesisState);
        message.claimRecords = [];
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.moduleAccountBalance = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.params = params_1.Params.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.claimRecords.push(claim_record_1.ClaimRecord.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseGenesisState);
        message.claimRecords = [];
        if (object.moduleAccountBalance !== undefined &&
            object.moduleAccountBalance !== null) {
            message.moduleAccountBalance = coin_1.Coin.fromJSON(object.moduleAccountBalance);
        }
        else {
            message.moduleAccountBalance = undefined;
        }
        if (object.params !== undefined && object.params !== null) {
            message.params = params_1.Params.fromJSON(object.params);
        }
        else {
            message.params = undefined;
        }
        if (object.claimRecords !== undefined && object.claimRecords !== null) {
            for (var _i = 0, _a = object.claimRecords; _i < _a.length; _i++) {
                var e = _a[_i];
                message.claimRecords.push(claim_record_1.ClaimRecord.fromJSON(e));
            }
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseGenesisState);
        message.claimRecords = [];
        if (object.moduleAccountBalance !== undefined &&
            object.moduleAccountBalance !== null) {
            message.moduleAccountBalance = coin_1.Coin.fromPartial(object.moduleAccountBalance);
        }
        else {
            message.moduleAccountBalance = undefined;
        }
        if (object.params !== undefined && object.params !== null) {
            message.params = params_1.Params.fromPartial(object.params);
        }
        else {
            message.params = undefined;
        }
        if (object.claimRecords !== undefined && object.claimRecords !== null) {
            for (var _i = 0, _a = object.claimRecords; _i < _a.length; _i++) {
                var e = _a[_i];
                message.claimRecords.push(claim_record_1.ClaimRecord.fromPartial(e));
            }
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.moduleAccountBalance !== undefined &&
            (obj.moduleAccountBalance = message.moduleAccountBalance
                ? coin_1.Coin.toJSON(message.moduleAccountBalance)
                : undefined);
        message.params !== undefined &&
            (obj.params = message.params ? params_1.Params.toJSON(message.params) : undefined);
        if (message.claimRecords) {
            obj.claimRecords = message.claimRecords.map(function (e) {
                return e ? claim_record_1.ClaimRecord.toJSON(e) : undefined;
            });
        }
        else {
            obj.claimRecords = [];
        }
        return obj;
    }
};
var globalThis = (function () {
    if (typeof globalThis !== "undefined")
        return globalThis;
    if (typeof self !== "undefined")
        return self;
    if (typeof window !== "undefined")
        return window;
    if (typeof global !== "undefined")
        return global;
    throw "Unable to locate global object";
})();
