// @stargazefi/api/lib/generated/cosmos/slashing/v1beta1/slashing.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
exports.Params = exports.ValidatorSigningInfo = exports.protobufPackage = void 0;
/* eslint-disable */
var Long = require("long");
var duration_1 = require("../../../google/protobuf/duration");
var timestamp_1 = require("../../../google/protobuf/timestamp");
var minimal_1 = require("protobufjs/minimal");
exports.protobufPackage = "cosmos.slashing.v1beta1";
var baseValidatorSigningInfo = {
    address: "",
    startHeight: Long.ZERO,
    indexOffset: Long.ZERO,
    tombstoned: false,
    missedBlocksCounter: Long.ZERO
};
exports.ValidatorSigningInfo = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (!message.startHeight.isZero()) {
            writer.uint32(16).int64(message.startHeight);
        }
        if (!message.indexOffset.isZero()) {
            writer.uint32(24).int64(message.indexOffset);
        }
        if (message.jailedUntil !== undefined) {
            timestamp_1.Timestamp.encode(toTimestamp(message.jailedUntil), writer.uint32(34).fork()).ldelim();
        }
        if (message.tombstoned === true) {
            writer.uint32(40).bool(message.tombstoned);
        }
        if (!message.missedBlocksCounter.isZero()) {
            writer.uint32(48).int64(message.missedBlocksCounter);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseValidatorSigningInfo);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.address = reader.string();
                    break;
                case 2:
                    message.startHeight = reader.int64();
                    break;
                case 3:
                    message.indexOffset = reader.int64();
                    break;
                case 4:
                    message.jailedUntil = fromTimestamp(timestamp_1.Timestamp.decode(reader, reader.uint32()));
                    break;
                case 5:
                    message.tombstoned = reader.bool();
                    break;
                case 6:
                    message.missedBlocksCounter = reader.int64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseValidatorSigningInfo);
        if (object.address !== undefined && object.address !== null) {
            message.address = String(object.address);
        }
        else {
            message.address = "";
        }
        if (object.startHeight !== undefined && object.startHeight !== null) {
            message.startHeight = Long.fromString(object.startHeight);
        }
        else {
            message.startHeight = Long.ZERO;
        }
        if (object.indexOffset !== undefined && object.indexOffset !== null) {
            message.indexOffset = Long.fromString(object.indexOffset);
        }
        else {
            message.indexOffset = Long.ZERO;
        }
        if (object.jailedUntil !== undefined && object.jailedUntil !== null) {
            message.jailedUntil = fromJsonTimestamp(object.jailedUntil);
        }
        else {
            message.jailedUntil = undefined;
        }
        if (object.tombstoned !== undefined && object.tombstoned !== null) {
            message.tombstoned = Boolean(object.tombstoned);
        }
        else {
            message.tombstoned = false;
        }
        if (object.missedBlocksCounter !== undefined &&
            object.missedBlocksCounter !== null) {
            message.missedBlocksCounter = Long.fromString(object.missedBlocksCounter);
        }
        else {
            message.missedBlocksCounter = Long.ZERO;
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseValidatorSigningInfo);
        if (object.address !== undefined && object.address !== null) {
            message.address = object.address;
        }
        else {
            message.address = "";
        }
        if (object.startHeight !== undefined && object.startHeight !== null) {
            message.startHeight = object.startHeight;
        }
        else {
            message.startHeight = Long.ZERO;
        }
        if (object.indexOffset !== undefined && object.indexOffset !== null) {
            message.indexOffset = object.indexOffset;
        }
        else {
            message.indexOffset = Long.ZERO;
        }
        if (object.jailedUntil !== undefined && object.jailedUntil !== null) {
            message.jailedUntil = object.jailedUntil;
        }
        else {
            message.jailedUntil = undefined;
        }
        if (object.tombstoned !== undefined && object.tombstoned !== null) {
            message.tombstoned = object.tombstoned;
        }
        else {
            message.tombstoned = false;
        }
        if (object.missedBlocksCounter !== undefined &&
            object.missedBlocksCounter !== null) {
            message.missedBlocksCounter = object.missedBlocksCounter;
        }
        else {
            message.missedBlocksCounter = Long.ZERO;
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.address !== undefined && (obj.address = message.address);
        message.startHeight !== undefined &&
            (obj.startHeight = (message.startHeight || Long.ZERO).toString());
        message.indexOffset !== undefined &&
            (obj.indexOffset = (message.indexOffset || Long.ZERO).toString());
        message.jailedUntil !== undefined &&
            (obj.jailedUntil =
                message.jailedUntil !== undefined
                    ? message.jailedUntil.toISOString()
                    : null);
        message.tombstoned !== undefined && (obj.tombstoned = message.tombstoned);
        message.missedBlocksCounter !== undefined &&
            (obj.missedBlocksCounter = (message.missedBlocksCounter || Long.ZERO).toString());
        return obj;
    }
};
var baseParams = { signedBlocksWindow: Long.ZERO };
exports.Params = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (!message.signedBlocksWindow.isZero()) {
            writer.uint32(8).int64(message.signedBlocksWindow);
        }
        if (message.minSignedPerWindow.length !== 0) {
            writer.uint32(18).bytes(message.minSignedPerWindow);
        }
        if (message.downtimeJailDuration !== undefined) {
            duration_1.Duration.encode(message.downtimeJailDuration, writer.uint32(26).fork()).ldelim();
        }
        if (message.slashFractionDoubleSign.length !== 0) {
            writer.uint32(34).bytes(message.slashFractionDoubleSign);
        }
        if (message.slashFractionDowntime.length !== 0) {
            writer.uint32(42).bytes(message.slashFractionDowntime);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseParams);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.signedBlocksWindow = reader.int64();
                    break;
                case 2:
                    message.minSignedPerWindow = reader.bytes();
                    break;
                case 3:
                    message.downtimeJailDuration = duration_1.Duration.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.slashFractionDoubleSign = reader.bytes();
                    break;
                case 5:
                    message.slashFractionDowntime = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseParams);
        if (object.signedBlocksWindow !== undefined &&
            object.signedBlocksWindow !== null) {
            message.signedBlocksWindow = Long.fromString(object.signedBlocksWindow);
        }
        else {
            message.signedBlocksWindow = Long.ZERO;
        }
        if (object.minSignedPerWindow !== undefined &&
            object.minSignedPerWindow !== null) {
            message.minSignedPerWindow = bytesFromBase64(object.minSignedPerWindow);
        }
        if (object.downtimeJailDuration !== undefined &&
            object.downtimeJailDuration !== null) {
            message.downtimeJailDuration = duration_1.Duration.fromJSON(object.downtimeJailDuration);
        }
        else {
            message.downtimeJailDuration = undefined;
        }
        if (object.slashFractionDoubleSign !== undefined &&
            object.slashFractionDoubleSign !== null) {
            message.slashFractionDoubleSign = bytesFromBase64(object.slashFractionDoubleSign);
        }
        if (object.slashFractionDowntime !== undefined &&
            object.slashFractionDowntime !== null) {
            message.slashFractionDowntime = bytesFromBase64(object.slashFractionDowntime);
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseParams);
        if (object.signedBlocksWindow !== undefined &&
            object.signedBlocksWindow !== null) {
            message.signedBlocksWindow = object.signedBlocksWindow;
        }
        else {
            message.signedBlocksWindow = Long.ZERO;
        }
        if (object.minSignedPerWindow !== undefined &&
            object.minSignedPerWindow !== null) {
            message.minSignedPerWindow = object.minSignedPerWindow;
        }
        else {
            message.minSignedPerWindow = new Uint8Array();
        }
        if (object.downtimeJailDuration !== undefined &&
            object.downtimeJailDuration !== null) {
            message.downtimeJailDuration = duration_1.Duration.fromPartial(object.downtimeJailDuration);
        }
        else {
            message.downtimeJailDuration = undefined;
        }
        if (object.slashFractionDoubleSign !== undefined &&
            object.slashFractionDoubleSign !== null) {
            message.slashFractionDoubleSign = object.slashFractionDoubleSign;
        }
        else {
            message.slashFractionDoubleSign = new Uint8Array();
        }
        if (object.slashFractionDowntime !== undefined &&
            object.slashFractionDowntime !== null) {
            message.slashFractionDowntime = object.slashFractionDowntime;
        }
        else {
            message.slashFractionDowntime = new Uint8Array();
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.signedBlocksWindow !== undefined &&
            (obj.signedBlocksWindow = (message.signedBlocksWindow || Long.ZERO).toString());
        message.minSignedPerWindow !== undefined &&
            (obj.minSignedPerWindow = base64FromBytes(message.minSignedPerWindow !== undefined
                ? message.minSignedPerWindow
                : new Uint8Array()));
        message.downtimeJailDuration !== undefined &&
            (obj.downtimeJailDuration = message.downtimeJailDuration
                ? duration_1.Duration.toJSON(message.downtimeJailDuration)
                : undefined);
        message.slashFractionDoubleSign !== undefined &&
            (obj.slashFractionDoubleSign = base64FromBytes(message.slashFractionDoubleSign !== undefined
                ? message.slashFractionDoubleSign
                : new Uint8Array()));
        message.slashFractionDowntime !== undefined &&
            (obj.slashFractionDowntime = base64FromBytes(message.slashFractionDowntime !== undefined
                ? message.slashFractionDowntime
                : new Uint8Array()));
        return obj;
    }
};
var globalThis = (function () {
    if (typeof globalThis !== "undefined")
        return globalThis;
    if (typeof self !== "undefined")
        return self;
    if (typeof window !== "undefined")
        return window;
    if (typeof global !== "undefined")
        return global;
    throw "Unable to locate global object";
})();
var atob = globalThis.atob ||
    (function (b64) { return globalThis.Buffer.from(b64, "base64").toString("binary"); });
function bytesFromBase64(b64) {
    var bin = atob(b64);
    var arr = new Uint8Array(bin.length);
    for (var i = 0; i < bin.length; ++i) {
        arr[i] = bin.charCodeAt(i);
    }
    return arr;
}
var btoa = globalThis.btoa ||
    (function (bin) { return globalThis.Buffer.from(bin, "binary").toString("base64"); });
function base64FromBytes(arr) {
    var bin = [];
    for (var i = 0; i < arr.byteLength; ++i) {
        bin.push(String.fromCharCode(arr[i]));
    }
    return btoa(bin.join(""));
}
function toTimestamp(date) {
    var seconds = numberToLong(date.getTime() / 1000);
    var nanos = (date.getTime() % 1000) * 1000000;
    return { seconds: seconds, nanos: nanos };
}
function fromTimestamp(t) {
    var millis = t.seconds.toNumber() * 1000;
    millis += t.nanos / 1000000;
    return new Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new Date(o);
    }
    else {
        return fromTimestamp(timestamp_1.Timestamp.fromJSON(o));
    }
}
function numberToLong(number) {
    return Long.fromNumber(number);
}
if (minimal_1.util.Long !== Long) {
    minimal_1.util.Long = Long;
    minimal_1.configure();
}
