// @stargazefi/api/lib/generated/stargaze/alloc/v1beta1/params.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
exports.Params = exports.DistributionProportions = exports.WeightedAddress = exports.protobufPackage = void 0;
/* eslint-disable */
var minimal_1 = require("protobufjs/minimal");
exports.protobufPackage = "publicawesome.stargaze.alloc.v1beta1";
var baseWeightedAddress = { address: "", weight: "" };
exports.WeightedAddress = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.weight !== "") {
            writer.uint32(18).string(message.weight);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseWeightedAddress);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.address = reader.string();
                    break;
                case 2:
                    message.weight = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseWeightedAddress);
        if (object.address !== undefined && object.address !== null) {
            message.address = String(object.address);
        }
        else {
            message.address = "";
        }
        if (object.weight !== undefined && object.weight !== null) {
            message.weight = String(object.weight);
        }
        else {
            message.weight = "";
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseWeightedAddress);
        if (object.address !== undefined && object.address !== null) {
            message.address = object.address;
        }
        else {
            message.address = "";
        }
        if (object.weight !== undefined && object.weight !== null) {
            message.weight = object.weight;
        }
        else {
            message.weight = "";
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.address !== undefined && (obj.address = message.address);
        message.weight !== undefined && (obj.weight = message.weight);
        return obj;
    }
};
var baseDistributionProportions = {
    nftIncentives: "",
    developerRewards: ""
};
exports.DistributionProportions = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.nftIncentives !== "") {
            writer.uint32(10).string(message.nftIncentives);
        }
        if (message.developerRewards !== "") {
            writer.uint32(18).string(message.developerRewards);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseDistributionProportions);
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.nftIncentives = reader.string();
                    break;
                case 2:
                    message.developerRewards = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseDistributionProportions);
        if (object.nftIncentives !== undefined && object.nftIncentives !== null) {
            message.nftIncentives = String(object.nftIncentives);
        }
        else {
            message.nftIncentives = "";
        }
        if (object.developerRewards !== undefined &&
            object.developerRewards !== null) {
            message.developerRewards = String(object.developerRewards);
        }
        else {
            message.developerRewards = "";
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseDistributionProportions);
        if (object.nftIncentives !== undefined && object.nftIncentives !== null) {
            message.nftIncentives = object.nftIncentives;
        }
        else {
            message.nftIncentives = "";
        }
        if (object.developerRewards !== undefined &&
            object.developerRewards !== null) {
            message.developerRewards = object.developerRewards;
        }
        else {
            message.developerRewards = "";
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.nftIncentives !== undefined &&
            (obj.nftIncentives = message.nftIncentives);
        message.developerRewards !== undefined &&
            (obj.developerRewards = message.developerRewards);
        return obj;
    }
};
var baseParams = {};
exports.Params = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = minimal_1.Writer.create(); }
        if (message.distributionProportions !== undefined) {
            exports.DistributionProportions.encode(message.distributionProportions, writer.uint32(10).fork()).ldelim();
        }
        for (var _i = 0, _a = message.weightedDeveloperRewardsReceivers; _i < _a.length; _i++) {
            var v = _a[_i];
            exports.WeightedAddress.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof Uint8Array ? new minimal_1.Reader(input) : input;
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = globalThis.Object.create(baseParams);
        message.weightedDeveloperRewardsReceivers = [];
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.distributionProportions = exports.DistributionProportions.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.weightedDeveloperRewardsReceivers.push(exports.WeightedAddress.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        var message = globalThis.Object.create(baseParams);
        message.weightedDeveloperRewardsReceivers = [];
        if (object.distributionProportions !== undefined &&
            object.distributionProportions !== null) {
            message.distributionProportions = exports.DistributionProportions.fromJSON(object.distributionProportions);
        }
        else {
            message.distributionProportions = undefined;
        }
        if (object.weightedDeveloperRewardsReceivers !== undefined &&
            object.weightedDeveloperRewardsReceivers !== null) {
            for (var _i = 0, _a = object.weightedDeveloperRewardsReceivers; _i < _a.length; _i++) {
                var e = _a[_i];
                message.weightedDeveloperRewardsReceivers.push(exports.WeightedAddress.fromJSON(e));
            }
        }
        return message;
    },
    fromPartial: function (object) {
        var message = __assign({}, baseParams);
        message.weightedDeveloperRewardsReceivers = [];
        if (object.distributionProportions !== undefined &&
            object.distributionProportions !== null) {
            message.distributionProportions = exports.DistributionProportions.fromPartial(object.distributionProportions);
        }
        else {
            message.distributionProportions = undefined;
        }
        if (object.weightedDeveloperRewardsReceivers !== undefined &&
            object.weightedDeveloperRewardsReceivers !== null) {
            for (var _i = 0, _a = object.weightedDeveloperRewardsReceivers; _i < _a.length; _i++) {
                var e = _a[_i];
                message.weightedDeveloperRewardsReceivers.push(exports.WeightedAddress.fromPartial(e));
            }
        }
        return message;
    },
    toJSON: function (message) {
        var obj = {};
        message.distributionProportions !== undefined &&
            (obj.distributionProportions = message.distributionProportions
                ? exports.DistributionProportions.toJSON(message.distributionProportions)
                : undefined);
        if (message.weightedDeveloperRewardsReceivers) {
            obj.weightedDeveloperRewardsReceivers = message.weightedDeveloperRewardsReceivers.map(function (e) { return (e ? exports.WeightedAddress.toJSON(e) : undefined); });
        }
        else {
            obj.weightedDeveloperRewardsReceivers = [];
        }
        return obj;
    }
};
var globalThis = (function () {
    if (typeof globalThis !== "undefined")
        return globalThis;
    if (typeof self !== "undefined")
        return self;
    if (typeof window !== "undefined")
        return window;
    if (typeof global !== "undefined")
        return global;
    throw "Unable to locate global object";
})();
